packages/mongo/test/DatabaseCollection.test.ts

```diff
@@ -233,4 +234,12 @@ describe("DatabaseCollection", () => {
         expect(find2).toHaveLength(1);
         expect(find2[0].id).toEqual("id1");
     });
+
+    test("should not allow to create duplicate entries for unique index", async function () {
+        const db = await getRandomCollection(["id"]);
+
+        await db.create({ id: "uniqueValue" });
+
+        await expect(db.create({ id: "uniqueValue" })).rejects.toThrow(/[Dd]uplicate key/);
```

Comment:
I dislike the fact that the error message varies depending on the database provider used. Therefore, I suggest catching the error in the Collection Providers and throwing a custom error instead. We could also add an error code to the custom error to make it easier to reliably catch, even if the error message changes in the future.

Reply:
But can't we at least just do it here now?

Reply:
We could, but why should we? It's important that an error is thrown, but we will not catch it in a specific way